### ISSUE TITLE
Navigation Bar title and back button color not changing with setThemeUsingPrimaryColor

### ISSUE THREAD

**Issue by avyayv** [ViccAlexander/Chameleon#138]
When I do  `setThemeUsingPrimaryColor(arrayOfColors[2] as! UIColor, withSecondaryColor: arrayOfColors[1] as! UIColor, andContentStyle: .Contrast)`, where arrayOfColors is an array of colors based on an image, I get the following for my Navigation Bar
(https://cloud.githubusercontent.com/assets/18251101/18456967/f249ce9e-7909-11e6-8cda-dbf1d47227c5.png), where the text isn't that great looking. I don't know if I'm doing something wrong, but the barbuttonitem is working fine so...

**Comment by bre7** [comment]
What was the expected result ?

Here's how Chameleon themes the Nav bar:
```objc
UIColor *contentColor = contentColor = ContrastColor(primaryColor, NO); // When .Contrast is used
[[UINavigationBar appearanceWhenContainedIn:[self class], nil] setBarTintColor:primaryColor];
[[UINavigationBar appearanceWhenContainedIn:[self class], nil] setTintColor:contentColor];
[[UINavigationBar appearanceWhenContainedIn:[self class], nil] setTitleTextAttributes:@{NSForegroundColorAttributeName:contentColor}];
[[UINavigationBar appearanceWhenContainedIn:[self class], nil] setShadowImage:[UIImage new]];
```

**Comment by ViccAlexander** [comment]
@avyayv Are you sure you're not overriding the tint color somewhere else in your code?

**Comment by avyayv** [comment]
Yes I'm sure I'm not, but the way I fixed my personal issue was that I used contrastColor.

**Comment by MinsungHuh** [comment]
I also not chaning with setThemeUsingprimaryColor, swift 3, iOS 10, Xcode 8.

**Comment by bre7** [comment]
Confirmed.

Steps to reproduce:
1. Use Demo project
2. Remove theme code inside AppDelegate
3. Add the following code to ViewController.swift
```swift
setThemeUsingPrimaryColor(.flatPlum,
                                  withSecondaryColor: .flatBlue,
                                  andContentStyle: .contrast)
```

Results:
* Pink color NavigationBar (set in Main.storyboard)
* Default tint color ("blue") Back button

Expected Results:
* NavigationBar should be colored with `.flatPlum`
* Back button should be colored with `white` (since we are using `.contrast` style)

**Comment by MinsungHuh** [comment]
<img width="623" alt="chameleon_application" src="https://cloud.githubusercontent.com/assets/8328382/19372815/3e759c40-91fa-11e6-947d-a0296dfcba50.png">
<img width="624" alt="chameleon_viewcontroller" src="https://cloud.githubusercontent.com/assets/8328382/19372822/4623421c-91fa-11e6-8af1-200301d35552.png">
![chameleon_result](https://cloud.githubusercontent.com/assets/8328382/19372868/8c63a94c-91fa-11e6-82f7-d06f32f62bb7.png)

It does not seem to be working properly.

**Comment by niclashedam** [comment]
Is there any update on this?

**Comment by j-pk** [comment]
This issue also occurs to the tintColors on SFSafariViewControllers.

**Comment by User2004** [comment]
Write following link code in appdelegate.m file for set navigation bar tint color, font color apply font family etc.

https://webappcodes.com/set-navigation-bar-theme-in-objective-c

**Hide navigation back button and set navigation title** 

https://webappcodes.com/set-navigation-bar-title-programmatically-and-hide-default-navigation-back-button-in-objective-c

**Set navigation left button programatically in objective c**

https://webappcodes.com/set-navigation-left-bar-button-in-objective-c
